Sources/Common/MultipartRouteType.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  MultipartRouteType.swift
+//  OperaSwift
+//
+//  Created by Miguel Revetria on 5/3/17.
+//
+//
+
+import Alamofire
+import Foundation
+import UIKit
+
+// MARK: - MultipartRequestType
+
+public struct MultipartData {
+
+    let data: Data
+    let fileName: String
+    let mimeType: String
+    let name: String
+
+}
+
+public protocol MultipartRouteType: RouteType {
+
+    var items: [MultipartData] { get }
+
+}
+
+public extension MultipartRouteType {
+
+    var method: HTTPMethod {
+        return .post
+    }
+
+}
+
+// MARK: - ImageUploadRouteType
+
+public enum ImageUploadEncoding {
+
+    case png
+    case jpeg(quality: CGFloat)
+
+    var fileExtension: String {
+        switch self {
+        case .png:
+            return "png"
+        case .jpeg:
+            return "jpg"
+        }
+    }
+
+    func encode(image: UIImage) -> Data? {
+        switch self {
+        case .png:
+            return UIImagePNGRepresentation(image)
+        case let .jpeg(quality):
+            return UIImageJPEGRepresentation(image, quality)
+        }
+    }
+
+}
+
+public protocol ImageUploadRouteType: MultipartRouteType {
+
+    var encoding: ImageUploadEncoding { get }
+    var image: UIImage { get set }
+    var fileName: String { get }
+    var imageName: String { get }
+    var mimeType: String { get }
+
```

Comment:
I agree with you about the `mimeType`. Due to this protocol allow you to send PNG or JPEG representations of images the mime type wouldn't be an option. Maybe you want to use `jpeg` instead of `jpg` but looks like the standard says that it should be jpeg (https://www.w3.org/Graphics/JPEG/)

The `fileName` is not the same case to me. It is something that may depend on the service implementation. For example, a custom implementation could save the image at disk using the fileName so you may want to use a timestamp as part of the name. What do you think?